File: autumn/human-readable-number/go/log10.go
package main

import (
   "fmt"
   "math"
)

func numberFormat(d float64) string {
   e := int(math.Log10(d)) / 3
   f := d / math.Pow10(e * 3)
   return fmt.Sprintf("%.3f", f) + []string{"", " k", " M", " G"}[e]
}

func main() {
   s := numberFormat(9012345678)
   println(s == "9.012 G")
}

File: autumn/human-readable-number/rb/log10.rb
def number_format(d)
   e = Math.log10(d).to_i / 3
   return '%.3f' % (d / 1e3 ** e) + ['', ' k', ' M', ' G'][e]
end

s = number_format(9012345678)
puts s == '9.012 G'
